src/pages/_app.tsx

```diff
@@ -1,16 +1,33 @@
 // Based on https://github.com/mui-org/material-ui/blob/master/examples/nextjs/pages/_app.js
 import '../../scripts/globals'
 import * as React from 'react'
-import PropTypes from 'prop-types'
 import Head from 'next/head'
-import { settings } from '~/api/__mock__'
+import type { AppProps as NextAppProps } from 'next/app'
+import type { NextPage } from 'next'
+import type { EmotionCache } from '@emotion/cache'
 import { RootProvider } from '~/contexts'
 import * as layoutVariants from '~/layouts'
 
-function App(props) {
-  const { Component, defaultLocale, emotionCache, locale, pageProps: pagePropsProp } = props
+export interface AppProps extends NextAppProps {
+  Component: NextPage<Page>
+  emotionCache?: EmotionCache
+  pageProps: {
+    defaultLocale: string
+    locale: string
+    headerColor?: string
+    headerMode?: string
+    layout?: keyof typeof layoutVariants
+    page: Page
+    settings: Record<string, unknown>
+    theme?: string
+  }
+}
+
+function App(props: AppProps) {
+  const { Component, emotionCache, pageProps: pagePropsProp } = props
 
-  const { headerColor, headerMode, layout, page, ...pageProps } = pagePropsProp
+  const { defaultLocale, headerColor, headerMode, layout, locale, page, ...pageProps } =
+    pagePropsProp
```

Comment:
Perhaps rename the "page" related props to something more understandable @alexanderflink